spec/controllers/home_page_spec.rb

```diff
@@ -1,13 +1,29 @@
 require 'rails_helper'
 
 RSpec.describe HomePageController do
-  describe "GET #index" do
+  describe 'GET #index' do
     let(:do_request) { get :index }
 
-    it "renders the index template" do
-      do_request
-      expect(response).to have_http_status(:ok)
-      expect(response).to render_template(:index)
+    context 'not logged in user' do
+      it 'redirects to WelcomeController' do
+        do_request
+        expect(response).to have_http_status 302
+        expect(response).to redirect_to controller: 'welcome', action: 'index'
+      end
+    end
+
+    context 'logged in user' do
+
+      user = FactoryBot.create(:user)
+      user.confirm
```

Comment:
I suggest wrapping what user creating into the `let`, consider:
```ruby
let(:user) do
  user = FactoryBot.create(:user)
  user.confirm
  user
end
```

Reply:
It seems like we'll create confirmed user quite often in the future. At least more often than unconfirmed. So my proposal is to create confirmed user by default and unconfirmed on demand (`FactoryBot.create(:user, :unconfirmed)`)